src/receive_imf.rs

```diff
@@ -677,12 +677,22 @@ pub(crate) async fn receive_imf_inner(
         let res = if mime_parser.incoming {
             handle_securejoin_handshake(context, &mut mime_parser, from_id)
                 .await
-                .context("error in Secure-Join message handling")?
+                .with_context(|| {
+                    format!(
+                        "error in Secure-Join '{}' message handling",
+                        mime_parser.get_header(HeaderDef::SecureJoin).unwrap_or("")
```

Comment:
Can `if let Some(...)` be used above instead of `mime_parser.get_header(HeaderDef::SecureJoin).is_some()` so we don't call `get_header` again? Or does borrow checker complain?